src/components/logger/index.ts

```diff
@@ -100,99 +94,105 @@ export async function probeLog({
     }
   }
 
-  await saveProbeRequestLog({
+  saveProbeRequestLog({
     probe,
     totalRequests,
     probeRes,
     alerts,
     error,
   })
+
+  return mLog
 }
 
 /**
  * notificationLog just prints notifications for the user and to persistent log (through history.ts)
- *
+ * @param {objct} contain notification fields
+ * @param {LogObjec} mLog to update and return
+ * @returns {LogObject} mLog is returned once updated
  */
-export async function notificationLog({
-  type,
-  alertMsg,
-  notification,
-  probe,
-}: {
-  probe: Probe
-  notification: Notification
-  type: 'NOTIFY-INCIDENT' | 'NOTIFY-RECOVER'
-  alertMsg: string
-}) {
+export function setNotificationLog(
+  {
+    type,
+  }: {
+    probe: Probe
+    notification: Notification
+    type: 'NOTIFY-INCIDENT' | 'NOTIFY-RECOVER'
+    alertMsg: string
+  },
+  mLog: LogObject
+): LogObject {
   let msg: string
-  mLogs.notification.flag = type
+
   switch (type) {
     case 'NOTIFY-INCIDENT':
       msg = 'service probably down'
       break
     case 'NOTIFY-RECOVER':
       msg = 'service is back up'
   }
-  mLogs.notification.flag = type
-  mLogs.notification.message[0] = msg
-  await saveNotificationLog(probe, notification, type, alertMsg)
```

Comment:
the `saveNotificationLog` now is not called anywhere

Reply:
added back!